pkg/runtime/runtime-manifest.go

```diff
@@ -100,13 +102,51 @@ func resolveCredential(cd manifest.CredentialDefinition) (string, error) {
 	}
 }
 
+// resolveBundleOutput will attempt to resolve a bundle output value if required
+// by string interpolation, e.g. via {{ bundle.outputs.foo }}
+// It first checks the outputs field on the RuntimeManifest struct and, if non-existent,
+// attempts to pull a value from the claim, if exists
+func (m *RuntimeManifest) resolveBundleOutput(output manifest.OutputDefinition) (string, error) {
+	if m.outputs[output.Name] != "" {
+		return m.outputs[output.Name], nil
+	}
+
+	if claimBytes, err := m.FileSystem.ReadFile(config.ClaimFilepath); err == nil {
```

Comment:
nit: How about we make a new function `loadClaim` and then run it from `Validate` (which is called once before any steps are run). Then we don't reparse the claim for each output reference.

Reply:
Definitely, good thinking.  This logic has been revised appropriately.